crates/red_knot_python_semantic/resources/mdtest/assignment/augmented.md

```diff
@@ -96,7 +96,7 @@ def _(flag: bool):
     f = Foo()
     f += "Hello, world!"
 
-    reveal_type(f)  # revealed: int | str
+    reveal_type(f)  # revealed: Unknown | int | str
```

Comment:
I'm not sure I understand where the new `Unknown` is coming from here? We are revealing the type of `f` locally within the scope, so it shouldn't be due to modifiability of `f`. And both `__add__` and `__iadd__` are declared in the body of `Foo` (because function definition statements are declarations), so I don't think either of those should be unioned with `Unknown`?

Reply:
`__iadd__` is possibly undeclared.

I'm not 100% clear on the semantics of `__add__` and `__iadd__`, so I suspect that you think that this possibly-undeclaredness should be absorbed by the definite-declaredness of `__add__`?

I'll look into that tomorrow.

Reply:
Hmm. I said off the top of my head that we should add `Unknown` to the union for possibly-undeclared, because the "declared type" in the undeclared path is `Unknown`. But now I'm wondering if that's just wrong. An external assignment that violates the possibly-declared type would effectively be creating a conflicting-declarations situation, and we currently error on conflicting declarations, so it seems like we should also error on that assignment. Which we would do, if we didn't union with `Unknown` for possibly-undeclared.

Reply:
That makes sense to me. To make it more concrete, my understanding is that we would always want to show an error in the last line here, because it is possible that `flag` is `True`, and in this case, we would violate the declaration.
```py
def _(flag: bool):
    class C:
        if flag:
            var: int

        var = 1

    C.var = "a"
```
we would always want to show an error in the last line.

I'll revert f4ca4b88c1530eae60df633d6d41a866e1f97d4b and add a new test.